firewood/src/manager.rs

```diff
@@ -262,32 +274,68 @@ impl RevisionManager {
 
         Ok(())
     }
-}
-
-impl RevisionManager {
-    pub fn add_proposal(&self, proposal: ProposedRevision) {
-        self.proposals.lock().expect("poisoned lock").push(proposal);
-    }
 
+    /// View the database at a specific hash.
+    /// To view the database at a specific hash involves a few steps:
+    /// 1. Try to find it in committed revisions.
+    /// 2. Try to find it in proposals.
+    /// 3. Try to find it in `RootStore`.
     pub fn view(&self, root_hash: HashKey) -> Result<ArcDynDbView, RevisionManagerError> {
-        // First try to find it in committed revisions
+        // 1. Try to find it in committed revisions.
         if let Ok(committed) = self.revision(root_hash.clone()) {
             return Ok(committed);
         }
 
-        // If not found in committed revisions, try proposals
+        // 2. Try to find it in proposals.
         let proposal = self
             .proposals
             .lock()
             .expect("poisoned lock")
             .iter()
             .find(|p| p.root_hash().as_ref() == Some(&root_hash))
-            .cloned()
-            .ok_or(RevisionManagerError::RevisionNotFound {
-                provided: root_hash,
-            })?;
+            .cloned();
 
-        Ok(proposal)
+        if let Some(proposal) = proposal {
+            return Ok(proposal);
+        }
+
+        // 3. Try to find it in `RootStore`.
+        let revision_addr =
+            self.root_store
+                .get(&root_hash)?
+                .ok_or(RevisionManagerError::RevisionNotFound {
+                    provided: root_hash.clone(),
+                })?;
+
+        let node_store = NodeStore::with_root(
+            root_hash.into_hash_type(),
+            revision_addr,
+            self.current_revision(),
+        );
+
+        Ok(Arc::new(node_store))
+    }
+}
+
+impl<T> RevisionManager<T> {
+    pub fn add_proposal(&self, proposal: ProposedRevision) {
+        self.proposals.lock().expect("poisoned lock").push(proposal);
+    }
+
+    pub fn all_hashes(&self) -> Vec<TrieHash> {
```

Comment:
This needs some more thought. Do we need to support fetching all hashes from the RootStore? This might be handy for debugging. If so, we probably don't want to return a `Vec` anymore, this should be an iterator.

Maybe this shouldn't be called `all_hashes`...

I think a TODO here would be sufficient, as this would be an API change.